Wrappers/Python/test/test_DataContainer.py

```diff
@@ -931,6 +931,170 @@ def test_axpby7(self):
+        b = ig.allocate(dtype=numpy.float32)         
+
+        d1.fill(numpy.asarray(numpy.arange(1,101).reshape(10,10), dtype=numpy.float32))
+        d2.fill(numpy.asarray(numpy.arange(1,101).reshape(10,10), dtype=numpy.float32))
+        a.fill(1.0/d1.as_array())                                                  
+        b.fill(-1.0/d2.as_array())   
+
+        out = ig.allocate(-1,dtype=numpy.float32)                                                 
+        # equals to 1 + -1 = 0
+        out = d1.sapyb(a,d2,b)
+        res = numpy.zeros_like(d1.as_array())
+        numpy.testing.assert_array_equal(res, out.as_array())
+
+        out.fill(0)
+        d1.sapyb(a,d2,b, out)
+        res = numpy.zeros_like(d1.as_array())
+        numpy.testing.assert_array_equal(res, out.as_array())
+
+    def test_sapyb_scalar_f(self):
+        # a,b scalar
+        ig = ImageGeometry(10,10)                                               
+        d1 = ig.allocate(1, dtype=numpy.float32)                                                     
+        d2 = ig.allocate(2, dtype=numpy.float32)   
+        a = 2.
+        b = -1.
+
+        out = ig.allocate(-1,dtype=numpy.float32)                                                 
+        # equals to 2*[1] + -1*[2] = 0
+        out = d1.sapyb(a,d2,b)
+        res = numpy.zeros_like(d1.as_array())
+        numpy.testing.assert_array_equal(res, out.as_array())
+
+        out.fill(0)
+        d1.sapyb(a,d2,b, out)
+        numpy.testing.assert_array_equal(res, out.as_array())
+
+        d1.sapyb(a,d2,b, out=d1)
+        numpy.testing.assert_array_equal(res, d1.as_array())
+
+        d1.fill(1)
+        d1.sapyb(a,d2,b, out=d2)
+        numpy.testing.assert_array_equal(res, d2.as_array())
+    
+    def test_sapyb_datacontainer_scalar_f(self):
+        #mix: a scalar and b DataContainer and a DataContainer and b scalar
+        ig = ImageGeometry(10,10)                                               
+        d1 = ig.allocate(1., dtype=numpy.complex64)                                                     
+        d2 = ig.allocate(2.,dtype=numpy.complex64)   
+        a = 2.+2j                                                
+        b = ig.allocate(-1.-1j, dtype=numpy.complex64)         
+
+
+        out = ig.allocate(-1,dtype=numpy.complex64)
+        # equals to (2+2j)*[1] + -(1+j)*[2] = 0
+        
+        out = d1.sapyb(a,d2,b)
+        res = ig.allocate(0, dtype=numpy.complex64)
+        numpy.testing.assert_array_equal(res.as_array(), out.as_array())
+
+        out.fill(-1)
+        d1.sapyb(a,d2,b, out)
+        numpy.testing.assert_array_equal(res.as_array(), out.as_array())
+
+        out = d2.sapyb(b,d1,a)
+        res = ig.allocate(0, dtype=numpy.complex64)
+        numpy.testing.assert_array_equal(res.as_array(), out.as_array())
+
+        out.fill(-1)
+        d2.sapyb(b,d1,a, out)
+        numpy.testing.assert_array_equal(res.as_array(), out.as_array())
+
+    def test_sapyb_scalar_c(self):
```

Comment:
What is the difference with  `test_sapyb_datacontainer_scalar_f`

Reply:
`test_sapyb_scalar_c` is for complex only scalar values of the parameters `a` and `b`. `test_sapyb_datacontainer_scalar_f` is for mixed scalar and `DataContainer` `a` and `b` in float.